File: cpp/21.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *head=new ListNode(0);
        ListNode *pCur=head;
        while(l1 && l2)
        {
            if(l1->val <l2->val)
            {
                pCur->next=new ListNode(l1->val);
                l1=l1->next;
            }
            else
            {
                pCur->next=new ListNode(l2->val);
                l2=l2->next;
            }
            pCur=pCur->next;
        }
        while(l1)
        {
            pCur->next=new ListNode(l1->val);
            l1=l1->next;
            pCur=pCur->next;
        }
        while(l2)
        {
            pCur->next=new ListNode(l2->val);
            l2=l2->next;
            pCur=pCur->next;
        }
        return head->next;
    }
};

File: golang/21.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    var nodeHead *ListNode = new(ListNode)
    var pCur *ListNode = nodeHead
    
    for {
        if list1 == nil || list2 == nil{
            break
        }
        if list1.Val < list2.Val {
            pCur.Next = new(ListNode)
            pCur = pCur.Next
            pCur.Val = list1.Val
            list1 = list1.Next
        } else{
            pCur.Next = new(ListNode)
            pCur = pCur.Next
            pCur.Val = list2.Val
            list2 = list2.Next
        }
    }
    
    for ;list1 != nil; {
        pCur.Next = new(ListNode)
        pCur = pCur.Next
        pCur.Val = list1.Val
        list1 = list1.Next
    }
    
    for ;list2 != nil; {
        pCur.Next = new(ListNode)
        pCur = pCur.Next
        pCur.Val = list2.Val
        list2 = list2.Next
    }
    return nodeHead.Next
}
